packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.m

```diff
@@ -218,7 +218,22 @@ - (NSAttributedString *)measurableAttributedText
 
 - (CGSize)sizeThatFitsMinimumSize:(CGSize)minimumSize maximumSize:(CGSize)maximumSize
 {
-  NSAttributedString *attributedText = [self measurableAttributedText];
+  NSMutableAttributedString *attributedText = [[self measurableAttributedText] mutableCopy];
+
+  /*
+  * The block below is responsible for setting the exact height of the view in lines
```

Comment:
Maybe a question for @sammy-SC, is this kosher with any sort of AttributedString layout caching we are doing? I.e. is there something we need to cache bust if layout behavior here varies based on more of the number of line props?